### PR TITLE
pkg/ns: refactored so that builds succeed on non-linux platforms

### PR DESCRIPTION
**Pull request by aaithal**
#### Summary

moved functions that depend on linux packages (sys/unix) into a separate file
and added nop methods with a build tag for non-linux platforms in a new file.

#### Testing Done
`./test` ran successfully, apart from the `macvlan Operations` suite. I tested this on Amazon Linux and there's no support for macvlan on that platform. The failure log is pasted next:

```
• Failure [0.034 seconds]
macvlan Operations
/home/ec2-user/workspace/src/github.com/containernetworking/cni/gopath/src/github.com/containernetworking/cni/plugins/main/macvlan/macvlan_test.go:192
  creates an macvlan link in a non-default namespace [It]
  /home/ec2-user/workspace/src/github.com/containernetworking/cni/gopath/src/github.com/containernetworking/cni/plugins/main/macvlan/macvlan_test.go:102

  Expected error:
      <*errors.errorString | 0xc42014a690>: {
          s: "failed to create macvlan: operation not supported",
      }
      failed to create macvlan: operation not supported
  not to have occurred

  /home/ec2-user/workspace/src/github.com/containernetworking/cni/gopath/src/github.com/containernetworking/cni/plugins/main/macvlan/macvlan_test.go:87
```

### CODE REVIEW

**Comment by dcbw** [comment]
Should ns_linux.go have a +build linux in it?

**Comment by squeed** [comment]
Nah, ending a file with `_$GOOS` is an implicit build tag.

File: pkg/ns/ns_unspecified.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !linux
+
+package ns
+
+func getCurrentThreadNetNSPath() string {
```
Comment (squeed) [inline pkg/ns/ns_unspecified.go]: This makes me a bit uncomfortable: this is used in `ns.Do()`. I know that it currently won't cause problems, but...
Could you refactor `Do()` to use `GetCurrentNS()`, move `GetCurrentNS()` to the per-os files, and move `getCurrentThreadNetNSPath` to `ns_linux.go`?

Comment (aaithal) [inline pkg/ns/ns_unspecified.go]: Yes, sure. That's cleaner than how I did it.

**Comment by aaithal** [comment]
@squeed I have updated the PR to address your comments, could you please take another look?

@dcbw as per @squeed's comments, the implicit tag should take care of it.

**Comment by squeed** [comment]
Wow, I'm surprised that `netlink` builds on non-linux :-).

The code changes LGTM. If anyone wants the commits to be squashed, that'd be reasonable.

**Comment by dcbw** [comment]
Updates LGTM.

**Pull request merged**